source/solvers/vof_scratch_data.cc

```diff
@@ -0,0 +1,57 @@
+#include <core/bdf.h>
+#include <core/sdirk.h>
+
+#include <solvers/vof_scratch_data.h>
+
+template <int dim>
+void
+VOFScratchData<dim>::allocate()
+{
+  // Initialize size of arrays
+  this->n_q_points = fe_values_fs.get_quadrature().size();
+  this->n_dofs     = fe_values_fs.get_fe().n_dofs_per_cell();
+
+  // Initialize arrays related to quadrature
+  this->JxW = std::vector<double>(n_q_points);
+
+  // Forcing term array
+  this->source = std::vector<double>(n_q_points);
+
+
+  // Initialize arrays related to velocity and pressure
+  this->velocities.first_vector_component = 0;
+  // Velocity
+  this->velocity_values          = std::vector<Tensor<1, dim>>(n_q_points);
+  this->velocity_gradient_values = std::vector<Tensor<2, dim>>(n_q_points);
+
+
+  this->present_phase_values = std::vector<double>(n_q_points);
+  this->phase_gradients      = std::vector<Tensor<1, dim>>(n_q_points);
+  this->phase_laplacians     = std::vector<double>(n_q_points);
+
+  // Velocity for BDF schemes
+  this->previous_phase_values =
+    std::vector<std::vector<double>>(maximum_number_of_previous_solutions(),
+                                     std::vector<double>(n_q_points));
+
+
+
+  // Velocity for SDIRK schemes
```

Comment:
is it used somewhere? If not, it can be removed. If it is written for later implementation of SDIRK schemes, please say so in the comment, like : Velocity for SDIRK schemes (to be implemented)